neqo-http3/src/priority.rs

```diff
@@ -0,0 +1,74 @@
+use crate::{HFrame, Header};
+use std::fmt;
+use std::io::Write;
+
+#[derive(Debug, Copy, Clone, PartialEq, Eq)]
+pub struct Priority {
+    urgency: u8,
+    incremental: bool,
+}
+
+impl Default for Priority {
+    fn default() -> Self {
+        Priority {
+            urgency: 3,
+            incremental: false,
+        }
+    }
+}
+
+impl Priority {
+    /// # Panics
+    /// If an invalid urgency (>7 is given)
+    pub fn new(urgency: u8, incremental: bool) -> Priority {
+        assert!(urgency < 8);
+        Priority {
+            urgency,
+            incremental,
+        }
+    }
+
+    /// Returns a header if required to send
+    pub fn header(self) -> Option<Header> {
+        match self {
+            Priority {
+                urgency: 3,
+                incremental: false,
+            } => None,
+            other => Some(Header::new("priority", format!("{}", other))),
+        }
+    }
+
+    pub fn encode_request_frame(self, stream_id: u64) -> HFrame {
+        let mut priority = Vec::new();
+        write!(priority, "{}", self).unwrap();
```

Comment:
I would copy *self* into the HFrame.  It's small enough and you then don't need to worry about copying a Vec instance around.

Reply:
Done. But unfortunately I don't know if I can get rid of the two allocations when encoding: https://github.com/mozilla/neqo/pull/1200/commits/e1c6398920b9989ef0031ae580e3528fc1c7990b